quickstart/src/main/java/com/microsoft/azure/kusto/quickstart/KustoSampleApp.java

```diff
@@ -0,0 +1,395 @@
+            Scanner scanner = new Scanner(System.in);
+            scanner.nextLine();
+        }
+    }
+
+    private static ConnectionStringBuilder generateConnectionString(String endpointUrl, String authenticationMode) {
+        ConnectionStringBuilder csb = null;
+        switch (authenticationMode) {
+            case "UserPrompt":
+                csb = ConnectionStringBuilder.createWithUserPrompt(endpointUrl);
+                break;
+            case "ManagedIdentity":
+                // Connect using the system or user provided managed identity (Azure service only)
+                // TODO - Config (Optional): Managed identity client id if you are using a User Assigned Managed Id
+                String clientId = System.getenv("MANAGED_IDENTITY_CLIENT_ID");
+                if (StringUtils.isBlank(clientId)) {
+                    csb = ConnectionStringBuilder.createWithAadManagedIdentity(endpointUrl);
+                } else {
+                    csb = ConnectionStringBuilder.createWithAadManagedIdentity(endpointUrl, clientId);
+                }
+                break;
+            case "AppKey":
+                // TODO - Config (Optional): App Id & tenant, and App Key to authenticate with
+                // For information about how to procure an AAD Application see: https://docs.microsoft.com/azure/data-explorer/provision-azure-ad-app
+                csb = ConnectionStringBuilder.createWithAadApplicationCredentials(endpointUrl, System.getenv("APP_ID"), System.getenv("APP_KEY"), System.getenv("APP_TENANT"));
+                break;
+            case "AppCertificate":
+                csb = createAppCertificateStringBuilder(endpointUrl);
+                break;
+            default:
+                die(String.format("Authentication mode %s is not supported", authenticationMode));
+        }
+
+        csb.setApplicationNameForTracing("KustoJavaSdkQuickStart");
+        return csb;
+    }
+
+    private static ConnectionStringBuilder createAppCertificateStringBuilder(String endpointUrl) {
+        String appId = System.getenv("APP_ID");
+        String tenantId = System.getenv("APP_TENANT");
+        String privateKeyFilePath = System.getenv("PRIVATE_KEY_PEM_FILE_PATH");
+        String publicCertFilePath = System.getenv("PUBLIC_CERT_FILE_PATH");
+
+        try {
+            PrivateKey privateKey = SecurityUtils.getPrivateKey(publicCertFilePath);
+            X509Certificate x509Certificate = SecurityUtils.getPublicCertificate(privateKeyFilePath);
+            return ConnectionStringBuilder.createWithAadApplicationCertificate(endpointUrl, appId, x509Certificate, privateKey, tenantId);
+        } catch (IOException | GeneralSecurityException e) {
+            die("Couldn't create ConnectionStringBuilder for app certificate authentication", e);
+            return null;
+        }
+    }
+
+    private static boolean execute(Client kustoClient, String databaseName, String query, ExecutionType executionType) {

```

Comment:
I don't like this merge of execute and control command.

Reply:
Why not?
Not only is all the code in this method identical, but also in the Java SDK there is only 1 `execute` command, used identically for both queries and control commands.
Also, I thought you liked this idea last time we spoke about it.

I think duplicating code makes it harder for new users to understand the code.

Reply:
It kind of feels odd.
Commands need to return success (printing is decorative)
Query needs to return results (printing is substitute for handling it)

Reply:
This makes it less reusable - user wise. 
He'd need to split the function to do something

Reply:
I don't see why they'd have to split it to use it. In the Java SDK, it's combined. The user sends all commands, whether control commands or queries, to this one method. To me, it seems simpler.

The underlying method for both returns the result table (`KustoOperationResult`) if successful, where the user can see details (details if control command which probably isn't interesting in most cases, and query result if query). To me, this consistency is intuitive. Splitting into distinct cases to me sounds unintuitive.

But it seems you see it the opposite way, so I'll split it.